src/mc-fle2-payload-iev-v2.c

```diff
@@ -322,9 +325,93 @@ void mc_FLE2IndexedEncryptedValueV2_destroy(mc_FLE2IndexedEncryptedValueV2_t *ie
     _mongocrypt_buffer_cleanup(&iev->DecryptedServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->ServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->S_KeyId);
+
+    if (iev->type != kTypeRange) {
+        for (int i = 0; i < iev->edge_count; i++) {
+            mc_FLE2TagAndEncryptedMetadataBlock_cleanup(&iev->metadata[i]);
+        }
+    } else {
+        // kTypeEquality only has one metadata block
```

Comment:
`assert(iev->edge_count == 1);`

Reply:
Addressed with consolidation of functions.